### PR TITLE
[RHS-222] - Course Overview: Floating Card

### PR DESCRIPTION
**Pull request by a4bello**
## Implementation Description
The PR implements the floating card functionality on the course overview page. The card displays various text on its button depending on different scenarios (as highlighted in the ticket). As specified in the ticket, the button activates different page links depending on a variety of scenarios. This PR implements them all. 

## Screenshots
image
image
image
image

## What should reviewers focus on?
- Ensuring the button text displays the correct text based on the different scenarios specified in the ticket specifications.
- Ensuring the links the page is redirected to when the button is clicked corresponds to the specifications of the ticket. 

## Testing Procedure
- What test(s) should we run to make sure your code works?
   - What ways could the input be wrong?
   - Is it secure? Could nefarious users access/break it?

## Things to Note
- Additional dependencies/libraries you've added?
- Things you'd want to know when coming back to the code after a few months

## Checklist
- [x] Ensure code follows style guide by running the linters
- [x] I have updated the documentation or deemed it unnecessary
- [x] I have linked the relevant issue in this PR
- [x] I have requested a review from the PL, as well as other dev(s) (and designers if necessary)

### CODE REVIEW

File: frontend/src/components/pages/CourseOverview.tsx [hunk 1]
```diff
@@ -1,28 +1,102 @@
...
+        if (moduleIndex === -1) {
+          //  Go to first Not Started moduleIndex (2nd Case)
+          moduleIndex = courseData?.course?.modules?.findIndex(
+            (module: ModuleType, index: number) =>
+              index >= moduleProgress?.moduleProgress?.length &&
+              module !== null,
+          );
+        }
+      }
+    }
```
Comment (jfdoming) [inline frontend/src/components/pages/CourseOverview.tsx:96]: At the end of all these if statements, `moduleIndex` may be null still if there was missing data. Could you add a check here and disable the "start" button if it fails?

**Review by jfdoming — COMMENTED** [review]
Great job on the logic here! Wondering about two things but looks good otherwise.

**Review by smilesthesad — APPROVED** [review]
Did a functional test and worked, code also looks good to me!

File: frontend/src/components/pages/CourseOverview.tsx [hunk 2]
```diff
@@ -1,28 +1,102 @@
...
+      moduleIndex = moduleProgress?.moduleProgress?.findIndex(
+        (module: ModuleProgressType) =>
+          module?.startedAt !== null && module?.completedAt === null,
+      );
+
+      if (moduleIndex === -1) {
+        //  Go to first Not Started moduleIndex (1st Case: Skip Case)
+        moduleIndex = moduleProgress?.moduleProgress?.findIndex(
+          (module: ModuleProgressType) =>
+            module?.startedAt === null && module?.completedAt === null,
```
[the remark this answers is not in the document]
Comment (a4bello) [reply frontend/src/components/pages/CourseOverview.tsx]: Done

File: frontend/src/components/pages/CourseOverview.tsx (hunk 1 above)
Comment (a4bello) [reply frontend/src/components/pages/CourseOverview.tsx:97]: Done

**Review by jfdoming — APPROVED** [review]
LGTM!

**Pull request merged**